src/main/java/org/jboss/pnc/cleaner/archiver/BuildArchiver.java

```diff
@@ -0,0 +1,437 @@
+import org.jboss.pnc.constants.Attributes;
+import org.jboss.pnc.dto.Build;
+import org.jboss.pnc.dto.DTOEntity;
+import org.jboss.pnc.dto.ProductMilestone;
+import org.jboss.pnc.dto.ProductMilestoneRef;
+import org.jboss.pnc.dto.ProductRef;
+import org.jboss.pnc.dto.ProductVersion;
+import org.jboss.pnc.enums.BuildStatus;
+import org.jboss.pnc.rest.api.parameters.BuildsFilterParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.core.Response;
+import java.io.BufferedReader;
+import java.time.temporal.ChronoField;
+import java.time.temporal.IsoFields;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.jboss.pnc.cleaner.archiver.ArchviedBuildRecord.ErrorGroup.INDY;
+import static org.jboss.pnc.cleaner.archiver.ArchviedBuildRecord.ErrorGroup.ND;
+import static org.jboss.pnc.cleaner.archiver.ArchviedBuildRecord.ErrorGroup.PNC;
+import static org.jboss.pnc.cleaner.archiver.ArchviedBuildRecord.ErrorGroup.PSI;
+import static org.jboss.pnc.cleaner.archiver.BuildArchiverStrings.*;
+
+/**
+ * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
+ */
+@ApplicationScoped
+public class BuildArchiver {
+    private final Logger logger = LoggerFactory.getLogger(BuildArchiver.class);
+    public static final String BUILD_ARCHIVED = "BUILD_ARCHIVED";
+    @Inject
+    @RestClient
+    FinalLogClient finalLogs;
+    @Inject
+    BuildClient buildClient;
+    @Inject
+    ProductMilestoneClient productMilestoneClient;
+    @Inject
+    ProductVersionClient productVersionClient;
+    @Inject
+    BuildArchiver self;
+    @ConfigProperty(name = "buildArchiver.trimmedLogMaxSize", defaultValue = "1000000")
+    private Integer trimmedLogMaxSize;
+
+    @Timed
+    @Scheduled(cron = "{buildArchiverScheduler.cron}")
+    public void archiveBuilds() {
+        logger.info("Archiving Builds ...");
+        Collection<Build> unverifiedBuilds = getUnarchivedBuilds().getAll();
+        logger.info("Found {} unverified builds.", unverifiedBuilds.size());
+        unverifiedBuilds.forEach(build -> archiveBuild(build));
+    }
+
+    @Timed
+    private RemoteCollection<Build> getUnarchivedBuilds() {
+        BuildsFilterParameters buildsFilterParameters = new BuildsFilterParameters();
+        buildsFilterParameters.setRunning(false);
+        List<String> attributes = Collections.singletonList("!" + BUILD_ARCHIVED);
+        try {
+            return buildClient.getAll(buildsFilterParameters, attributes);
+        } catch (RemoteResourceException e) {
+            logger.error("Cannot read remote builds.", e);
+            return RemoteCollection.empty();
+        }
+    }
+
+    @Timed
+    public void archiveBuild(Build build) {
+        long trimLogSize = 0;
+        if (build.getTemporaryBuild() && build.getStatus() != BuildStatus.SUCCESS) {
```

Comment:
What does this do? Does it limit the size of analyzed logs only for the temporary not successful builds? Why this filter?

Reply:
it will set a `trimLogSize` only for non-successful temporary builds, so only these builds will store the trimmed logs (in case the temporary build gets deleted). Other builds will have the trimLogSize 0 so no trimmed logs.

Reply:
Aaah ok, cool. So looking at the final statuses, temporary build with status FAILED, NO_REBUILD_REQUIRED, REJECTED, REJECTED_FAILED_DEPENDENCIES, CANCELLED, SYSTEM_ERROR will have their trimmed logs stored. Perfect.